### PR TITLE
Promise ajax

### PR DESCRIPTION
**Pull request by geekygrappler**
Wraps Ajax with a promise... Paves the way for `async/await`... paves the way to drop in fetch in the future maybe.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ module.exports = {
     'plugin:ember-suave/recommended'
   ],
   env: {
-    browser: true
+    browser: true,
+    es6: true
```
Comment (geekygrappler) [inline .eslintrc.js]: So there's no eslint error for `Promise`

File: addon/utils/promise-ajax.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default (options) => {
+  return new Promise((resolve, reject) => {
```
Comment (rwjblue) [inline addon/utils/promise-ajax.js]: I don’t think we can rely on native Promise (unless we don’t support things like older IE and whatnot). We might need to add `import { Promise } from 'rsvp’`...

File: addon/utils/promise-ajax.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export default (options) => {
```
Comment (rwjblue) [inline addon/utils/promise-ajax.js]: If this file is only used in our own tests, can you move it to tests/helpers/?

Comment (geekygrappler) [reply addon/utils/promise-ajax.js]: :+1:

File: addon/utils/promise-ajax.js (hunk 2 above)
Comment (geekygrappler) [reply addon/utils/promise-ajax.js]: Can rely on native Promise if we're only using this as a test helper? Or will people run the tests against IE? Not something I really know about.

Comment (jherdman) [reply addon/utils/promise-ajax.js]: I'll raise my hand here. I build my tests against IE11.

Comment (geekygrappler) [reply addon/utils/promise-ajax.js]: Thanks @jherdman. I'll use rsvp.

**Comment by geekygrappler** [comment]
Should be good to go now @rwjblue.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**